plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx

```diff
@@ -6,66 +6,84 @@ import { Button, Card, CardBody } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { createInterpolateElement, useState } from '@wordpress/element';
 import { Link } from '@woocommerce/components';
-import { PaymentIncentive } from '@woocommerce/data';
+import { PaymentIncentive, PaymentProvider } from '@woocommerce/data';
 
 /**
  * Internal dependencies
  */
 import { WC_ASSET_URL } from '~/utils/admin-settings';
 import './incentive-banner.scss';
 import { StatusBadge } from '~/settings-payments/components/status-badge';
+import { isIncentiveDismissedInContext } from '~/settings-payments/utils';
 
 interface IncentiveBannerProps {
 	/**
 	 * Incentive data.
 	 */
 	incentive: PaymentIncentive;
+	/**
+	 * Payment provider.
+	 */
+	provider: PaymentProvider;
 	/**
 	 * Callback used when an incentive is accepted.
 	 *
-	 * @param id   Plugin ID.
-	 * @param slug Plugin slug.
+	 * @param id Incentive ID.
 	 */
-	onAccept: ( id: string, slug: string ) => void;
+	onAccept: ( id: string ) => void;
 	/**
 	 * Callback to handle dismiss action.
 	 *
 	 * @param dismissUrl Dismiss URL.
 	 * @param context    The context in which the incentive is dismissed. (e.g. whether it was in a modal or banner).
 	 */
 	onDismiss: ( dismissUrl: string, context: string ) => void;
+	/**
+	 * Callback to setup the plugin.
+	 *
+	 * @param id   Extension ID.
+	 * @param slug Extension slug.
+	 */
+	setupPlugin: ( id: string, slug: string ) => void;
 }
 
 export const IncentiveBanner = ( {
 	incentive,
+	provider,
 	onDismiss,
 	onAccept,
+	setupPlugin,
 }: IncentiveBannerProps ) => {
 	const [ isSubmitted, setIsSubmitted ] = useState( false );
 	const [ isDismissed, setIsDismissed ] = useState( false );
 	const [ isBusy, setIsBusy ] = useState( false );
 
-	const incentiveContext = 'wc_settings_payments__banner';
-
 	const handleAccept = () => {
 		setIsBusy( true );
-		onAccept( 'woopayments', 'woocommerce-payments' );
+		onAccept( incentive.id );
+		setupPlugin( provider.id, provider.plugin.slug );
 		setIsBusy( false );
 		setIsSubmitted( true );
 	};
 
 	const handleDismiss = () => {
 		setIsBusy( true );
-		onDismiss( incentive._links.dismiss.href, incentiveContext );
+		onDismiss(
+			incentive._links.dismiss.href,
+			'wc_settings_payments__banner'
```

Comment:
maybe extract to constant as it's used in two places

Reply:
+1

Reply:
Thanks, good suggestion! I've updated it.